portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,43 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("There is no secret");
+    // For now, hard code messages in json
+    ArrayList<String> jsonData = new ArrayList<String>();
+    jsonData.add("Message 1");
+    jsonData.add("Message 2");
+    jsonData.add("Message 3");
+
+    String jsonString = convertToJson(jsonData);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(jsonString);
+  }
+
+  /*
+   * Manually build a JSON string upon request (doGet())
+   */ 
+  private String convertToJson(ArrayList<String> data) {
+      String json = "{";
```

Comment:
Let's use a StringBuilder here.
String is an immutable object. Every time you reassign a value to it (including with the use of += operator), it has to create a new String object. So with a data.size()==3, you'll be creating ~10 different String objects in this method. StringBuilder is mutable so you won't have that problem.